src/CodeGenerator/src/Generator/WaiterGenerator.php

```diff
@@ -214,17 +216,24 @@ private function getAcceptorPathBody(WaiterAcceptor $acceptor): string
 
     private function getAcceptorErrorBody(ErrorWaiterAcceptor $acceptor): string
     {
-        $checks = ['$exception !== null'];
+        $checks = [];
         $error = $acceptor->getError();
+        if (null !== $expected = $acceptor->getExpected()) {
```

Comment:
IMHO this should be done only when the Exception does not have `error` attributes (The `QueueExistsWaiter` should not change)

Moreover, this is always true, as `getError()` called above is extract from `expected` value.

Do understand how it works, the `expected=ResourceNotFoundException` refer to a shape:
```yaml
    "ResourceNotFoundException":{
      "type":"structure",
      "members":{
        "message":{"shape":"ErrorMessage"}
      },
      "exception":true
    },
```
for SQS, this shape looks like
```
    "BatchEntryIdsNotDistinct":{
      "type":"structure",
      "members":{
      },
      "error":{
        "code":"AWS.SimpleQueueService.BatchEntryIdsNotDistinct",
        "httpStatusCode":400,
        "senderFault":true
      },
      "exception":true
    },
```

Is suggest to add the method `ExceptionShape::hasError`
```php
    public function hasError(): bool
    {
        return isset($this->data['error']);
    }
```

and change your code
```php
if (!$error->hasError()) {
    $checks[] = '$exception->getAwsCode() === ' . \var_export($expected, true);
} else {
    if (null !== $code = $error->getCode()) {
        $checks[] = '$exception->getAwsCode() === ' . \var_export($code, true);
    }
    if (null !== $code = $error->getStatusCode()) {
        $checks[] = '$exception->getCode() === ' . \var_export($code, true);
    }
}

Reply:
Done.